metrics_test.go

```diff
@@ -0,0 +1,526 @@
+					Unit:        metricdata.UnitBytes,
+					Type:        metricdata.TypeCumulativeDistribution,
+				},
+				Resource: nil,
+				TimeSeries: []*metricdata.TimeSeries{
+					{
+						StartTime: timestampToTime(startTimestamp),
+						Points: []metricdata.Point{
+							{
+								Time: timestampToTime(endTimestamp),
+								Value: &metricdata.Distribution{
+									Count:                 1,
+									Sum:                   11.9,
+									SumOfSquaredDeviation: 0,
+									Buckets: []metricdata.Bucket{
+										{Count: 1}, {}, {}, {},
+									},
+									BucketOptions: &metricdata.BucketOptions{
+										Bounds: []float64{10, 20, 30, 40},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			want: []*monitoringpb.CreateTimeSeriesRequest{
+				{
+					Name: "projects/foo",
+					TimeSeries: []*monitoringpb.TimeSeries{
+						{
+							Metric: &googlemetricpb.Metric{
+								Type: "custom.googleapis.com/opencensus/with_metric_descriptor",
+							},
+							Resource: &monitoredrespb.MonitoredResource{
+								Type: "global",
+							},
+							Points: []*monitoringpb.Point{
+								{
+									Interval: &monitoringpb.TimeInterval{
+										StartTime: startTimestamp,
+										EndTime:   endTimestamp,
+									},
+									Value: &monitoringpb.TypedValue{
+										Value: &monitoringpb.TypedValue_DistributionValue{
+											DistributionValue: &distributionpb.Distribution{
+												Count:                 1,
+												Mean:                  11.9,
+												SumOfSquaredDeviation: 0,
+												BucketCounts:          []int64{0, 1, 0, 0, 0},
+												BucketOptions: &distributionpb.Distribution_BucketOptions{
+													Options: &distributionpb.Distribution_BucketOptions_ExplicitBuckets{
+														ExplicitBuckets: &distributionpb.Distribution_BucketOptions_Explicit{
+															Bounds: []float64{0, 10, 20, 30, 40},
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for i, tt := range tests {
+		tsl, err := se.metricToMpbTs(context.Background(), tt.in)
+		if tt.wantErr != "" {
+			if err == nil || !strings.Contains(err.Error(), tt.wantErr) {
+				t.Errorf("#%d: unmatched error. Got\n\t%v\nWant\n\t%v", i, err, tt.wantErr)
+			}
+			continue
+		}
+		if err != nil {
+			t.Errorf("#%d: unexpected error: %v", i, err)
+			continue
+		}
+
+		got := se.combineTimeSeriesToCreateTimeSeriesRequest(tsl)
+		if !reflect.DeepEqual(got, tt.want) {
+			// Our saving grace is serialization equality since some
+			// unexported fields could be present in the various values.
+			gj, wj := serializeAsJSON(got), serializeAsJSON(tt.want)
+			if gj != wj {
+				t.Errorf("#%d: Unmatched JSON\nGot:\n\t%s\nWant:\n\t%s", i, gj, wj)
+			}
+		}
+	}
+}
+
+func TestMetricDescriptorToMonitoringMetricDescriptor(t *testing.T) {
+	t.Skip("Metric schema has changed. Fix this test later.")
+
+	tests := []struct {
```

Comment:
Same, consider adding a test on built-in metrics.

Reply:
Let me look.

Reply:
@songy23 is it ok if I do this in another PR? the reason is it would require to create a generic api to provide a formatter like [GetMetricType](https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/blob/81dc1cded66aa4c95ef161923b61e41e17073f89/stackdriver.go#L205) that works for any data type (view/metric/etc).

Reply:
Yes, I'm good with leaving a TODO here and have a separate PR for it.